File: src/chipschallenge/blockreactions/CannotMove.java
package chipschallenge.blockreactions;

import chipschallenge.Block;

/**
 * Standard behavior for when things CANNOT move
 */
public class CannotMove extends NoSlipReaction {

    private CannotMove() {
    }
    private static CannotMove mInstance = null;

    public static synchronized CannotMove getInstance() {
        if (mInstance == null) {
            mInstance = new CannotMove();
        }
        return mInstance;
    }

    public void react(Block moving, Block standing) {
        // No reaction
    }

    public final boolean canMove(Block moving, Block standing) {
        return false;
    }

}

File: ChipsChallenge.Shared/Blockreactions/CannotMove.cs
namespace ChipsChallenge.Shared.Blockreactions
{
    /// <summary>
    /// Standard behavior for when things CANNOT move
    /// </summary>
    public class CannotMove : NoSlipReaction
    {
        private static CannotMove instance;

        private CannotMove()
        {
        }

        public static CannotMove Instance
        {
            get
            {
                lock (typeof(CannotMove))
                {
                    return instance ?? (instance = new CannotMove());
                }
            }
        }

        public override void React(Block moving, Block standing)
        {
        }

        public override bool canMove(Block moving, Block standing)
        {
            return false;
        }
    }
}